lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -192,22 +192,52 @@ static Value buildUnitNormalCdf(OpBuilder &b, Location &loc, Value x) {
 static Value createLinalgPayloadCalculationForElementwiseOp(
     OpBuilder &b, Location loc, TypeConverter *converter,
     ValueRange payloadArgs, Operation *op, ArrayRef<Value> operands) {
-  if (isa<AtenTanhOp>(op))
-    return b.create<math::TanhOp>(loc, payloadArgs[0]);
-  if (isa<AtenExpOp>(op))
-    return b.create<math::ExpOp>(loc, payloadArgs[0]);
   if (isa<AtenFloorOp>(op))
     return b.create<math::FloorOp>(loc, payloadArgs[0]);
   if (isa<AtenCeilOp>(op))
     return b.create<math::CeilOp>(loc, payloadArgs[0]);
-  if (isa<AtenLogOp>(op))
-    return b.create<math::LogOp>(loc, payloadArgs[0]);
-  if (isa<AtenErfOp>(op))
-    return b.create<math::ErfOp>(loc, payloadArgs[0]);
-  if (isa<AtenSqrtOp>(op))
-    return b.create<math::SqrtOp>(loc, payloadArgs[0]);
-  if (isa<AtenRsqrtOp>(op))
-    return b.create<math::RsqrtOp>(loc, payloadArgs[0]);
+  if (auto tanh = dyn_cast<AtenTanhOp>(op)) {
+    Type dtype = converter->convertType(tanh.getType())
```

Comment:
the if body are identical for all the ops below. Can you use template function to reduce the duplicated code?
something like 
```
static template<typename TorchOpTy, typename MathOpTy>
createLinalgPayloadCalculationForElementwiseMathOp(OpBuilder &b, Location loc, TypeConverter *converter,
    ValueRange payloadArgs, Operation *op)
```